core/blockstm/dag.go

```diff
@@ -80,7 +100,7 @@ func (d DAG) Report(out func(string)) {
 		rootIds = append(rootIds, k)
 	}
 
-	sort.Ints(roots)
+	roots, rootIds = bubbleSort(roots, rootIds)
```

Comment:
Not sure if this is the best way to sort this. Can we create a map (roots -> rootIds), sort roots, and re-create a sorted rootId slace based on the sorted roots and the map?

Reply:
Yes sure, will do this.